apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx

```diff
@@ -431,22 +445,36 @@ export class DetailViewContents extends React.Component {
 
   renderStatusSelect = () => {
     const selectControl = (
-      <FormControl
-        componentClass="select"
-        disabled={this.state.locked || !this.state.editing}
-        title={this.state.locked && "The status of this application has been locked"}
-        value={this.state.status}
-        onChange={this.handleStatusChange}
-        style={styles.statusSelect}
-      >
-        {
-          Object.keys(this.statuses).map((status, i) => (
-            <option value={status} key={i}>
-              {this.statuses[status]}
-            </option>
-          ))
-        }
-      </FormControl>
+      <div>
+        <FormControl
+          componentClass="select"
+          disabled={this.state.locked || !this.state.editing}
+          title={this.state.locked && "The status of this application has been locked"}
+          value={this.state.status}
+          onChange={this.handleStatusChange}
+          style={styles.statusSelect}
+        >
+          {
+            Object.keys(this.statuses).map((status, i) => (
+              <option value={status} key={i}>
+                {this.statuses[status]}
+              </option>
+            ))
+          }
+        </FormControl>
+        <ConfirmationDialog
```

Comment:
If we're going to prevent users from assigning the application status to these states without an assigned workshop, then this needs to be verified server side as well.